cumulusci/salesforce_api/metadata.py

```diff
@@ -426,6 +426,10 @@ def _build_envelope_start(self):
                 run_tests=run_tests,
                 api_version=self.api_version,
             )
+        else:
+            # if there is a valid use-case for this branch, please
+            # add a unit test
+            assert self.package_zip, "package_zip should be specified"
```

Comment:
I don't think there is a valid use case. I think it would make more sense to assert that package_zip is not empty in ApiDeploy's constructor.